libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Decoding/Decoder.hs

```diff
@@ -758,14 +758,15 @@ decodeMapLikeEnforceNoDuplicates ::
   Decoder s (Maybe Int) ->
   Decoder s (k, v) ->
   Decoder s (Map.Map k v)
-decodeMapLikeEnforceNoDuplicates decodeLenOrIndef =
+decodeMapLikeEnforceNoDuplicates decodeLenOrIndef decodeKv = do
+  void decodeTagMaybe
   -- We first decode into a list because most of the time the encoded Map will be in sorted
   -- order and there is a nice optimization on the `Map.fromList` that can take advantage of
   -- that fact. In case when encoded data is not sorted the penalty of going through a list
   -- is insignificant.
-  decodeListLikeEnforceNoDuplicates decodeLenOrIndef (:) $ \xs ->
+  decodeListLikeEnforceNoDuplicates decodeLenOrIndef (:) (\xs ->
```

Comment:
I have doubts about whether to add this: here, or in the decoder that uses this function, or the opposite, even lower. 

Since it's enforcing no duplicates, it's  conceivable that other types deserialized with this will be specified to have the tag. 
On the other hand, we also have `decodeListLikeEnforceNoDuplicates`, so maybe the tag could be there. However, the latter is used for `decodeSetLikeEnforceNoDuplicates`, which has a mandatory tag.

I could add it in `decodeListLikeEnforceNoDuplicates` and parameterize whether to have it or not. 

What do you think is best @lehins ?

Reply:
Nope. Please revert any changes to this function.

This function is named `decodeListLikeEnforceNoDuplicates`, which has nothing to do with sets and in fact is used with an `OMap`, which cannot have a tag for a prefix

Another note is: do not use `decodeTagMaybe`, because it will allow any tag. We need to allow 258 tag, which means you can use `allowTag setTag` for doing that

Reply:
`decodeSetLikeEnforceNoDuplicates` does not have a mandatory tag, it allows a tag, which means it will consume 258 tag if it is present and it will fail if the tag is differs from 258

Reply:
Indeed, I misread `allowTag` thinking it was mandatory, but I see my mistake, thanks!